test-skripsi/test-improvement.tsx

```diff
@@ -23,11 +23,25 @@
 	}
 
 	getAllData(): Array<T> {
-		return this.data;
+		return [...this.data];
 	}
 
 	sortData(compareFunction: (a: T, b: T) => number): void {
-		this.data.sort(compareFunction);
+		if (this.data.length > 1000) {
+			this.data.sort(compareFunction);
+		} else {
+			this.insertionSort(compareFunction);
+		}
+	}
+
+	private insertionSort(compareFunction: (a: T, b: T) => number): void {
+		for (let i = 1; i < this.data.length; i++) {
+			let j = i;
+			while (j > 0 && compareFunction(this.data[j - 1], this.data[j]) > 0) {
+				[this.data[j - 1], this.data[j]] = [this.data[j], this.data[j - 1]];
+				j--;
+			}
+		}
```

Comment:
The implementation of `insertionSort` is clean and follows the algorithm's standard approach. For maintainability, consider adding a brief comment describing the algorithm's time complexity and why it's preferred for smaller datasets.